lib/system.nim

```diff
@@ -1754,11 +1754,12 @@ proc `$`*(x: int64): string {.magic: "Int64ToStr", noSideEffect.}
   ## The stringify operator for an integer argument. Returns `x`
   ## converted to a decimal string.
 
-when not defined(nimscript):
-  when not defined(JS) and hasAlloc:
+when not defined(nimscript) and not defined(JS) and hasAlloc:
     proc `$` *(x: uint64): string {.noSideEffect.}
       ## The stringify operator for an unsigned integer argument. Returns `x`
       ## converted to a decimal string.
+else:
+    proc `$` *(x: uint32): string {.magic: "Int64ToStr", noSideEffect.}
```

Comment:
@Araq, could you elaborate please? At least if you're not gonna let this change in ;)